src/containers/channelDetail/ChannelDetailContainer.jsx

```diff
@@ -35,6 +39,21 @@ class ChannelDetailContainer extends Component {
     }
   }
 
+  getSelectedSuggestion() {
+    // TODO: this should be treated as an url param in the future so we can avoid having to save this to session storage
+    //  we should have an url like /channelDetail/<token_address>/<partner_address>
+    if (this.props.selectedSuggestion) {
+      return this.props.selectedSuggestion;
+    } else {
+      const selectedSuggestion = sessionStorage.getItem("tempSelectedSuggestion");
+      if (selectedSuggestion) {
+        return JSON.parse(selectedSuggestion);
+      } else {
+        return null;
+      }
+    }
+  }
```

Comment:
I'd preffer something as following to prevent that many if-elses:
```suggestion
    if (this.props.selectedSuggestion) {
      return this.props.selectedSuggestion;
    }
    const selectedSuggestion = sessionStorage.getItem("tempSelectedSuggestion");
    if (selectedSuggestion) {
      return JSON.parse(selectedSuggestion);
    }
    return null;
  }
```

But do as you prefer